xformers/triton/k_softmax.py

```diff
@@ -158,8 +151,8 @@ def _softmax_backward(
     if causal:
         io_mask = io_mask & (k <= n)
 
-    g = tl.load(grad_out_ptrs, mask=io_mask, other=float(0))
-    o = tl.load(out_ptrs, mask=io_mask, other=float(0))
+    g = tl.load(grad_out_ptrs, mask=io_mask, other=float(0)).to(tl.float32)
```

Comment:
promote to float32 within the kernel, which doesn't change the arithmetics (max and exp were already fp32), and doesn´t touch the bandwidth (cast back to the correct type on the way out, and was of the correct type on the way in). Benchmarks are untouched